core/object/object.cpp

```diff
@@ -2097,10 +2097,10 @@ Object::~Object() {
 	// Disconnect signals that connect to this object.
 	while (connections.size()) {
 		Connection c = connections.front()->get();
-		bool disconnected = c.signal.get_object()->_disconnect(c.signal.get_name(), c.callable, true);
-		if (unlikely(!disconnected)) {
-			// If the disconnect has failed, abandon the connection to avoid getting trapped in an infinite loop here.
-			connections.pop_front();
+		Object *obj = c.callable.get_object();
+		bool disconnected = false;
+		if (likely(obj)) {
+			disconnected = c.signal.get_object()->_disconnect(c.signal.get_name(), c.callable, true);
```

Comment:
You're not doing anything with `disconnected`, so this raises a warning.

Reply:
Oh, right. It's no surprise I've kept messing up this PR as my last work hour today has been a waterfall of mistakes. 😅 I'll fix it and push.